pkg/loop/internal/relayer/pluginprovider/ext/median/test/median.go

```diff
@@ -93,6 +104,12 @@ func (s staticMedianFactoryServer) NewMedianFactory(ctx context.Context, provide
 		return nil, fmt.Errorf("NewMedianFactory: juelsPerFeeCoinDataSource does not equal a static test juels per fee coin data source implementation: %w", err)
 	}
 
+	err = s.gasPriceSubunitsDataSource.Evaluate(ctx, gasPriceSubunitsDataSource)
+	// allow for testing 0 value with the same staticMedianFactoryServer (only defined once)
+	if err != nil && !strings.HasSuffix(err.Error(), "got 0") {
```

Comment:
i'd have to rework a lot of logic if I'd want to create a staticMedianFactoryServer with 0 static data source value. Since this is just a test assertion i allow for 0 to come back as a valid value during the evaluation phase

Reply:
this is quite brittle. at minimium i would make a named error at the return site and use error.Is here rather than string matching

Reply:
sounds good, I can do that